baseplate/experiments/providers/base.py

```diff
@@ -0,0 +1,24 @@
+
+class ExperimentInterface(object):
```

Comment:
My thinking behind creating `Interface` classes for these is that the goal of this system is to be able to support different experiment providers that are constructed from a single factory method so I wanted to try to provide a basic interface that you could rely on.  If you don't like having this kind of thing around, I'm fine with removing it or waiting to add it until we _actually_ have different experiment types.